### PR TITLE
adding unit tests for broker package

### PR DESCRIPTION
**Pull request by shawn-hurley**
#### Describe what this PR does and why we need it:
Example of how to implement tests for the broker package.

#### Changes proposed in this pull request
 - test cases for job_state_subscriber moves to 81.7%
 - test cases for jobs moves to 70%
 - broker.go moves to 7.7%

### CODE REVIEW

File: pkg/broker/broker_test.go [hunk 1]
```diff
@@ -17,24 +17,211 @@
...
+func TestGetServiceInstance(t *testing.T) {
+	u := uuid.NewUUID()
+	errIsNotFound := fmt.Errorf("is not found")
+	unknownError := fmt.Errorf("is not found")
+	testCases := []struct {
+		name            string
+		dao             *mocks.Dao
+		config          Config
+		addExpectations func(*mocks.Dao)
+		shouldErr       error
```
Comment (maleck13) [inline pkg/broker/broker_test.go]: nit: I think most of tests like this use ```ExpectError``` not a big deal though

Comment (maleck13) [reply pkg/broker/broker_test.go]: ignore it is name the same usage

File: pkg/broker/broker_test.go [hunk 2]
```diff
@@ -17,24 +17,211 @@
...
+					},
+				}, nil)
+			},
+		},
+		{
+			name:   "get service instance error is not found",
+			dao:    new(mocks.Dao),
+			config: Config{},
+			addExpectations: func(d *mocks.Dao) {
+				d.On("GetServiceInstance", u.String()).Return(nil, unknownError)
```
Comment (maleck13) [inline pkg/broker/broker_test.go]: just trying to understand. Is this specifying the behavior when the GetServiceInstance is called or is it the expected result?

File: pkg/broker/jobs_test.go [hunk 3]
```diff
@@ -269,6 +271,190 @@ func TestApbJobRun(t *testing.T) {
...
+				if len(messages) != 2 {
+					return fmt.Errorf("expected 2 job messages")
+				}
+				first := messages[0]
+				if first.State.State != apb.StateInProgress {
+					return fmt.Errorf("unexpected first message contents")
+				}
+
+				second := messages[1]
+				log.Infof("second: %#v", second)
```
Comment (maleck13) [inline pkg/broker/jobs_test.go]: do we want to keep the log message?

File: pkg/broker/jobs_test.go [hunk 4]
```diff
@@ -269,6 +271,190 @@ func TestApbJobRun(t *testing.T) {
...
```
Comment (maleck13) [inline pkg/broker/jobs_test.go]: same might want to remove

File: pkg/broker/broker_test.go (hunk 2 above)
Comment (shawn-hurley) [reply pkg/broker/broker_test.go]: I think the answer is both.

https://github.com/stretchr/testify#mock-package 

```go
// TestSomething is an example of how to use our test object to
// make assertions about some target code we are testing.
func TestSomething(t *testing.T) {

  // create an instance of our test object
  testObj := new(MyMockedObject)

  // setup expectations
  testObj.On("DoSomething", 123).Return(true, nil)

  // call the code we are testing
  targetFuncThatDoesSomethingWithObj(testObj)

  // assert that the expectations were met
  testObj.AssertExpectations(t)
```

**Comment by jmrodri** [comment]
broker test coverage increased from 27.9% to 32.5%

**Pull request merged**